pkg/validation/validate_cluster.go

```diff
@@ -219,11 +219,19 @@ func (v *ValidationCluster) validateNodes(cloudGroups map[string]*cloudinstances
 			node := member.Node
 
 			if node == nil {
-				v.addError(&ValidationError{
-					Kind:    "Machine",
-					Name:    member.ID,
-					Message: fmt.Sprintf("machine %q has not yet joined cluster", member.ID),
-				})
+				nodeExpectedToJoin := true
+				if cloudGroup.InstanceGroup.Spec.Role == kops.InstanceGroupRoleBastion {
+					// bastion nodes don't join the cluster
+					nodeExpectedToJoin = false
+				}
+
+				if nodeExpectedToJoin {
```

Comment:
Why don't we have?

```go
if cloudGroup.InstanceGroup.Spec.Role != kops.InstanceGroupRoleBastion {
  v.addError(&ValidationError{
						Kind:    "Machine",
						Name:    member.ID,
						Message: fmt.Sprintf("machine %q has not yet joined cluster", member.ID),
					})
}
```

Confused why we have an extra if ...

Reply:
Because we expect more roles in future, and I find this code to be more self-descriptive (at the expense of being longer, admittedly)

Reply:
yes more flexiable ... k